wordpress-mobile/WordPress-iOS
# Adds dynamic type support to media permissions prompt

This PR improves dynamic type support in the new media permissions alert. To do this, we allow the label and buttons in the permissions header view to resize according to the current content size category, and if the category is an accessibility category (very large sizes) we hide the info icon and position the buttons vertically instead of horizontally to maximize the available space.

There's a small associated MediaPicker PR to allow the collection view to invalidate the header when the trait collection changes. This allows it to correctly resize if needed. https://github.com/wordpress-mobile/MediaPicker-iOS/pull/380

**Example screenshots**

|   |   |   |
|---|---|---|
| ![Simulator Screen Shot - iPhone 13 Pro - 2022-01-20 at 15 57 35](https://user-images.githubusercontent.com/4780/150405803-33bf0c28-b084-400f-baa6-5b4e0cb59bff.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-01-20 at 15 57 29](https://user-images.githubusercontent.com/4780/150405829-ee5e7e83-266c-4f58-8053-e932abddf130.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-01-20 at 15 57 12](https://user-images.githubusercontent.com/4780/150405837-6a1a71e6-0939-457b-b231-fdba0ea22f38.png) |

There are two very large sizes that slightly truncate the second button label, like so:

<img src="https://user-images.githubusercontent.com/4780/150408293-9ace9cea-73e1-4528-9bbb-b8d07e270d8a.png" width=300>

However, fixing this by allowing wrapping of the button text then caused issues with other sizes, so I've opted to leave this as-is for now.

### To test

* Build and run (the simulator is possibly easiest to test this)
* Navigate to Media
* If you haven't already chosen media permissions, choose to select specific photos. If you have already given permissions, head to System Settings > WordPress > Photos and choose selected images
* In Media, tap + and then Choose From My Device to bring up the device media picker. You should see the permissions prompt at the top once you've selected an album
* Using the X in Xcode, switch between different content sizes and check that the permissions alert prompt looks okay

This PR also disables the system prompt to change media permissions:

![IMG_9233](https://user-images.githubusercontent.com/4780/150408851-16f7554e-a775-4556-803d-24e826b320f4.jpg)

After performing the above steps, quit the app, then relaunch it and navigate back to Choose From My Device. Ensure you don't see the above prompt.

## Regression Notes

1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
